src/System.Private.CoreLib/shared/System/Threading/Timer.cs

```diff
@@ -55,14 +55,13 @@ private static TimerQueue[] CreateTimerQueues()
         #region interface to native timer
 
         private bool _isTimerScheduled;
-        private int _currentTimerStartTicks;
+        private long _currentTimerStartTicks;
         private uint _currentTimerDuration;
 
         private bool EnsureTimerFiresBy(uint requestedDuration)
         {
             // The VM's timer implementation does not work well for very long-duration timers.
-            // See kb 950807.
-            // So we'll limit our native timer duration to a "small" value.
+            // See kb 950807. So we limit our native timer duration to a "small" value.
```

Comment:
Nit: This kb article does not exist anymore (https://support.microsoft.com/en-us/help/950807 returns "page not found"), so mentioning it in a comment is not helpful.